File: client/golang/types/timeout.go
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */

package types

type Timeout struct {
	TaskId   string  `index:"0" json:"task_id" xml:"task_id" yaml:"task_id"`
	Binding  *Topic  `index:"5" json:"binding" xml:"binding" yaml:"binding"`
	Status   int64   `index:"10" json:"status" xml:"status" yaml:"status"`
	CreateAt int64   `index:"15" json:"create_at" xml:"create_at" yaml:"create_at"`
	InvokeAt int64   `index:"20" json:"invoke_at" xml:"invoke_at" yaml:"invoke_at"`
	Entity   *Entity `index:"25" json:"entity" xml:"entity" yaml:"entity"`
}

File: client/java/src/main/java/com/be/mesh/client/struct/Timeout.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package com.be.mesh.client.struct;

import com.be.mesh.client.annotate.Index;
import lombok.Data;

import java.io.Serializable;

/**
 * @author coyzeng@gmail.com
 */
@Data
public class Timeout implements Serializable {

    private static final long serialVersionUID = 6375030576735413376L;

    /**
     * Timeout task id.
     */
    @Index(value = 0, name = "task_id")
    private String taskId;

    /**
     * The timeout binding keys.
     */
    @Index(5)
    private Topic binding;

    /**
     * Timeout event status. &1=1 expired, &2=2 removed, &4=4 stopped.
     * <p>
     * If the timeout is stopped. If stopped will be pushed into stopped queue.
     */
    @Index(10)
    private long status;

    /**
     * Timeout initial time.
     */
    @Index(value = 15, name = "create_at")
    private long createAt;

    /**
     * Timeout expect time.
     */
    @Index(value = 20, name = "invoke_at")
    private long invokeAt;

    /**
     * Timeout payload entity
     */
    @Index(25)
    private Entity entity;

}
